File: server/src/main/java/com/github/util/GraphWrapper.java
package com.github.util;

import com.graphhopper.GHRequest;
import com.graphhopper.GraphHopperAPI;
import com.graphhopper.PathWrapper;
import kotlin.Pair;

import java.util.Locale;

public class GraphWrapper {

    private final GraphHopperAPI graph;

    public GraphWrapper(GraphHopperAPI graph) {
        this.graph = graph;
    }

    public PathWrapper simplePath(Pair<Double, Double> origin, Pair<Double, Double> target) {
        GHRequest req = new GHRequest(origin.component1(), origin.component2(), target.component1(), target.component2())
                .setWeighting("fastest")
                .setVehicle("car").setLocale(Locale.US);
        return graph.route(req).getBest();
    }
}

File: server/src/main/kotlin/com/github/util/GraphWrapper.kt
package com.github.util

import com.graphhopper.GHRequest
import com.graphhopper.GraphHopper
import com.graphhopper.ResponsePath

import java.util.Locale

/**
 * A Simple Wrapper to the GraphHopperAPI class.
 * @param graph A GraphHopperAPI instance.
 */
class GraphWrapper(private val graph: GraphHopper) {

    /**
     * Generates a PathWrapper containing the best route between origin and target points.
     * @param origin
     * @param target
     * @return
     */
    fun simplePath(origin: Pair<Double, Double>, target: Pair<Double, Double>): ResponsePath {
        val req = GHRequest(origin.first, origin.second, target.first, target.second)
                .setProfile("car_shortest")
                .setLocale(Locale.US)
        return graph.route(req).best
    }
}